<launch>
	<param name="use_sim_time" value="true"/>
	
    <node pkg="rosbag" type="play" name="player" args="--clock /home/amsl/bagfiles/2018-09-15-16-11-46.bag -s 250"/>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find infant_recognition)/../config/tsukuba2018.rviz"/>
	
	<include file="$(find velodyne_pointcloud)/launch/32e_points.launch"/>
	
	<node pkg="infant_recognition" type="tf_broadcaster" name="tf_broadcaster"/>

	<node pkg="infant_recognition" type="pc_transformer" name="velodyne2localmap"/>

	<node pkg="infant_recognition" type="occupancygrid_lidar" name="occupancygrid_lidar" output="screen"/>
	<node pkg="infant_recognition" type="occupancygrid_store" name="occupancygrid_store" output="screen"/>
	
	<node pkg="infant_recognition" type="grass2point" name="grass2grid"/>
	
	<node pkg="infant_recognition" type="pc_transformer" name="zed_grass2localmap">
		<remap from="/velodyne_points" to="/zed_grasspoints"/>
		<remap from="/velodyne_points/transformed" to="/zed_grasspoints/transformed"/>
	</node>
	<node pkg="infant_recognition" type="pc_transformer" name="zed_road2localmap">
		<remap from="/velodyne_points" to="/zed_roadpoints"/>
		<remap from="/velodyne_points/transformed" to="/zed_roadpoints/transformed"/>
	</node>
	
	<node pkg="infant_recognition" type="occupancygrid_zed" name="occupancygrid_zed" output="screen"/>
	
	<node pkg="infant_recognition" type="longest_path_detection" name="longest_path_detection" output="screen"/>

</launch>
